### PR TITLE
Nexus event load memory consumption

### LINKED ISSUE #1433: Memory consumption when loading event data
When loading event Nexus files using `sc.neutron.load` the memory consumption can become very high temporarily. We have previously worked on improving this, ensuring temporary workspaces are deleted as soon as possible.

One obvious culprit is https://github.com/scipp/scipp/blob/c3787ac3290bb6459b6b537197b3ace2d0b6dd96/python/src/scipp/compat/mantid.py#L670-L673, which currently makes a full copy of the loaded event table, but it is not clear if that is the main issue.

- Is there any memory we can free earlier so the peak consumption can be reduced?
- Should we change `sc.bins` to support moving/stealing from the input buffer?

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #1433.

There were actually two independent issues, note the two peaks:

Before:
before

After:
after

### CODE REVIEW

File: python/src/scipp/compat/mantid.py [hunk 1]
```diff
@@ -619,10 +619,12 @@ def convert_EventWorkspace_to_data_array(ws,
                         shape=[n_event],
                         unit=unit,
                         dtype=sc.dtype.float64)
-    weights = sc.Variable(dims=['event'],
-                          unit=data_unit,
-                          values=np.ones(n_event, dtype=np.float32),
-                          variances=np.ones(n_event, dtype=np.float32))
+    weights = sc.broadcast(sc.Variable(value=1.0,
```
Comment (OwenArnold) [inline python/src/scipp/compat/mantid.py:622]: How exactly does broadcast help with the memory profile?

File: python/src/scipp/compat/mantid.py [hunk 2]
```diff
@@ -673,7 +675,7 @@ def convert_EventWorkspace_to_data_array(ws,
     coords_labs_data["data"] = sc.bins(begin=begins,
                                        end=ends,
                                        dim='event',
-                                       data=events)
+                                       data=detail.move(events))
```
Comment (OwenArnold) [inline python/src/scipp/compat/mantid.py:678]: This follows through OK

File: python/src/scipp/compat/mantid.py (hunk 1 above)
Comment (SimonHeybrock) [reply python/src/scipp/compat/mantid.py:622]: Instead of creating a numpy array of ones and making a copy into scipp, we directly allow in scipp.

**Review by OwenArnold — CHANGES_REQUESTED** [review]
Otherwise looks good. 

If this also reduces the overall memory profile for the converters, could you please review the psutil settings in test_mantid.py, because that might open the possibility to run those tests on the azure hosted vms.

**Comment by SimonHeybrock** [comment]
> Otherwise looks good.
> 
> If this also reduces the overall memory profile for the converters, could you please review the psutil settings in test_mantid.py, because that might open the possibility to run those tests on the azure hosted vms.

It did *not* help there.... but I found the (unrelated) culprit and fixed it, so I could reduce requirements to 8 GByte (actual memory used by tests seems to stay just under 4 GByte).

**Review by OwenArnold — APPROVED** [review]
Nice work!

**Pull request merged**